python_modules/dagster/dagster/_core/definitions/asset_daemon_cursor.py

```diff
@@ -28,14 +22,81 @@
 
 from .asset_graph import AssetGraph
 from .partition import (
-    PartitionsDefinition,
     PartitionsSubset,
 )
 
 if TYPE_CHECKING:
     from dagster._core.instance import DynamicPartitionsStore
 
 
+class AssetDaemonAssetCursor(NamedTuple):
+    """Convenience class to represent the state of an individual asset being handled by the daemon.
+    In the future, this will be serialized as part of the cursor.
+    """
+
+    asset_key: AssetKey
+    latest_storage_id: Optional[int]
+    latest_evaluation_timestamp: Optional[float]
+    latest_evaluation: Optional[AutoMaterializeAssetEvaluation]
+    materialized_requested_or_discarded: Optional[bool] = None
```

Comment:
maybe is_materialized_requested_or_discarded to clearly differentiate from materialized_requested_or_discarded_subset

Reply:
Is this going to be replaced by `AssetPartitions` / `AssetSubset`?

Reply:
@sryza yes indeed -- just updated this PR to do that, and was able to shave off some more lines of code in the process (amazing how much code we have hanging around to deal with these situations)